src/tooltip/tooltip.js

```diff
@@ -400,13 +401,34 @@ angular.module('ui.bootstrap.tooltip', ['ui.bootstrap.position', 'ui.bootstrap.s
               ttScope.popupDelay = !isNaN(delay) ? delay : options.popupDelay;
             }
 
+            // hide tooltips/popovers when click away
+            function bodyHideTooltipBind(e) {
+              var target_elem = angular.element(e.target);
+              if (e.target != element[0] && !target_elem.hasClass('popover') && !target_elem.hasClass('popover-inner') && !target_elem.hasClass('popover-title') && !target_elem.hasClass('popover-content') && !target_elem.hasClass('tooltip') && !target_elem.hasClass('tooltip-inner')) {
```

Comment:
The check makes sure you're not clicking on a popover or tooltip itself, allowing the user to select text in the popover or tooltip. Otherwise, if you clicked on anything that wasn't the popover or tooltip element/link, it would close the popover. These classes are part of bootstrap.